File: develop/AsyncBreakfastVB/AsyncBreakfastVB/TestAsync.vb
Imports NUnit.Framework

<TestFixture>
Public Class TestAsync
    <TestCase>
    Public Shared Sub CheckCoffees(coffeeToPour As Integer)
        Try
            Assert.AreEqual(coffeeToPour, Coffee.coffeesPoured)
        Catch ex As AssertionException
            Console.Write(ex.Message)
        End Try
    End Sub

    <TestCase>
    Public Shared Sub CheckEggs(eggsToBoil As Integer)
        Try
            Assert.AreEqual(eggsToBoil, Egg.eggsBoiled)
        Catch ex As AssertionException
            Console.Write(ex.Message)
        End Try
    End Sub

    <TestCase>
    Public Shared Sub CheckBacon(baconToFry As Integer)
        Try
            Assert.AreEqual(baconToFry, Bacon.baconFried)
        Catch ex As AssertionException
            Console.Write(ex.Message)
        End Try
    End Sub

    <TestCase>
    Public Shared Sub CheckToasts(toastsToMake As Integer)
        Try
            Assert.AreEqual(toastsToMake, Toast.toastsMade)
        Catch ex As AssertionException
            Console.Write(ex.Message)
        End Try
    End Sub
End Class

File: develop/DevExpress/AsyncBreakfastC/AsyncBreakfastC/TestAsync.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using NUnit.Framework;
using AsyncBreakfastC.Model;
using Moq;

namespace AsyncBreakfastC
{
    [TestFixture]
    public class TestAsync
    {
        [TestCase]
        public static void CheckCoffees(int coffeeToPour)
        {
            try { Assert.AreEqual(coffeeToPour, Coffee.coffeesPoured); }
            catch (AssertionException ex) { Console.Write(ex.Message); }
        }

        [TestCase]
        public static void CheckEggs(int eggsToBoil)
        {
            try { Assert.AreEqual(eggsToBoil, Egg.eggsBoiled); }
            catch (AssertionException ex) { Console.Write(ex.Message); }
        }

        [TestCase]
        public static void CheckBacon(int baconToFry)
        {
            try { Assert.AreEqual(baconToFry, Bacon.baconFried); }
            catch (AssertionException ex) { Console.Write(ex.Message); }
        }

        [TestCase]
        public static void CheckToasts(int toastsToMake)
        {
            try { Assert.AreEqual(toastsToMake, Toast.toastsMade); }
            catch (AssertionException ex) { Console.Write(ex.Message); }
        }
    }
}
